iphone/Classes/MediaModule.m

```diff
@@ -1755,6 +1775,150 @@ - (void)handleTrimmedVideo:(NSURL *)theURL withDictionary:(NSDictionary *)dictio
+                           }];
+    } else if ([result.itemProvider canLoadObjectOfClass:UIImage.class]) {
+      [result.itemProvider loadObjectOfClass:UIImage.class
+                           completionHandler:^(__kindof id<NSItemProviderReading> _Nullable object, NSError *_Nullable error) {
+                             if (!error) {
+                               if (!imageArray) {
+                                 imageArray = [[NSMutableArray alloc] init];
+                               }
+                               TiBlob *media = [[[TiBlob alloc] initWithImage:(UIImage *)object] autorelease];
+                               [imageArray addObject:@{ @"media" : media, @"mediaType" : (NSString *)kUTTypeImage, @"success" : @(YES), @"code" : @(0) }];
+                             } else {
+                               [imageArray addObject:@{ @"error" : error.description, @"code" : @(error.code), @"success" : @(NO), @"mediaType" : (NSString *)kUTTypeImage }];
+                               DebugLog(@"[ERROR] Failed to load image- %@ .", error.description);
+                             }
+                             dispatch_group_leave(group);
+                           }];
+    } else if ([result.itemProvider hasItemConformingToTypeIdentifier:UTTypeMovie.identifier]) {
+      [result.itemProvider loadItemForTypeIdentifier:UTTypeMovie.identifier
+                                             options:nil
+                                   completionHandler:^(__kindof id<NSSecureCoding> _Nullable item, NSError *_Null_unspecified error) {
+                                     // Bug from Apple : https://developer.apple.com/forums/thread/652695
+                                     if (!error) {
+                                       if (!videoArray) {
+                                         videoArray = [[NSMutableArray alloc] init];
+                                       }
+                                       NSURL *url = (NSURL *)item;
+                                       TiBlob *media = [[[TiBlob alloc] initWithFile:[url path]] autorelease];
+                                       if ([media mimeType] == nil) {
+                                         [media setMimeType:@"video/mpeg" type:TiBlobTypeFile];
+                                       }
+                                       [videoArray addObject:@{ @"media" : media, @"mediaType" : (NSString *)kUTTypeMovie, @"success" : @(YES), @"code" : @(0) }];
+                                     } else {
+                                       [videoArray addObject:@{ @"error" : error.description, @"code" : @(error.code), @"success" : @(NO), @"mediaType" : (NSString *)kUTTypeMovie }];
+                                       DebugLog(@"[ERROR] Failed to load video- %@ .", error.description);
+                                     }
+                                     dispatch_group_leave(group);
+                                   }];
```

Comment:
There seem to be a lot of duplicate code logic here. Maybe move to own handler class / methods to clean up the business logic around this.